src/java/htsjdk/samtools/fastq/FastqRecord.java

```diff
@@ -24,18 +24,23 @@
 package htsjdk.samtools.fastq;
 
 import java.io.Serializable;
+import java.util.Objects;
 
 /**
  * Represents a fastq record, fairly literally, i.e. without any conversion.
  */
 public class FastqRecord implements Serializable {
     private static final long serialVersionUID = 1L;
     private final String seqHeaderPrefix;
-    private final String seqLine;
+    private final String seqLine;            //not null, not empty
```

Comment:
according to the FASTQ spec that @nh13 pointed out, these are not null and not empty. So I enforce it here and use that fact to simplify a bunch of code in this class. Tests get updated accordingly